set/entries.go

```diff
@@ -160,12 +160,12 @@ func (e *Entry) InodeStoragePath() string {
 
 // setTypeForNoInode sets our type based on the given dirent's Type, for the
 // case that the the dirent has no Inode (is missing or is a directory).
-func (e *Entry) setTypeForNoInode(dirent *walk.Dirent) {
-	if dirent.Type == os.ModeIrregular {
+func (e *Entry) setTypeForNoInode(dirent *Dirent) {
+	if dirent.Mode == os.ModeIrregular {
```

Comment:
I was thinking Dirent methods for all of these kinds of comparisons as well. So in addition to the IsDIr() method you added, IsIrregualr() etc.